tests/pytorch_tests/model_tests/feature_models/bn_folding_test.py

```diff
@@ -13,30 +13,102 @@
 # limitations under the License.
 # ==============================================================================
 import torch
+import numpy as np
+from model_compression_toolkit.core.pytorch.utils import set_model, to_torch_tensor, \
+    torch_tensor_to_numpy
 from tests.pytorch_tests.model_tests.base_pytorch_test import BasePytorchTest
 
 """
 This test checks the BatchNorm folding feature, plus adding a residual connection.
 """
 class BNFoldingNet(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, test_layer, fold_applied):
         super(BNFoldingNet, self).__init__()
-        self.conv1 = torch.nn.Conv2d(3, 3, kernel_size=1, stride=1)
-        self.bn = torch.nn.BatchNorm2d(3)
+        self.conv1 = test_layer
+        self.fold_applied = fold_applied
+        self.bn = torch.nn.BatchNorm2d(test_layer.out_channels)
 
     def forward(self, inp):
-        x = self.conv1(inp)
-        x = self.bn(x)
+        x1 = self.conv1(inp)
+        x = self.bn(x1)
         x = torch.relu(x)
-        return x + inp
+        if not self.fold_applied:
+            x = x + x1
+        return x
 
 
 class BNFoldingNetTest(BasePytorchTest):
     """
     This test checks the BatchNorm folding feature, plus adding a residual connection.
     """
-    def __init__(self, unit_test, float_reconstruction_error=1e-6):
+    def __init__(self, unit_test, test_layer, fold_applied=True, float_reconstruction_error=1e-6):
         super().__init__(unit_test, float_reconstruction_error)
+        self.test_layer = test_layer
+        self.fold_applied = fold_applied
+
+    def create_feature_network(self, input_shape):
+        return BNFoldingNet(self.test_layer, self.fold_applied)
+
+    def get_tpc(self):
+        return {'no_quantization': super().get_tpc()['no_quantization']}
+
+    def get_core_configs(self):
+        return {'no_quantization': super().get_core_configs()['no_quantization']}
+
+    def compare(self, quantized_models, float_model, input_x=None, quantization_info=None):
+        set_model(float_model)
+        quant_model = quantized_models['no_quantization']
+        set_model(quant_model)
+        out_float = torch_tensor_to_numpy(float_model(*input_x))
+        out_quant = torch_tensor_to_numpy(quant_model(*input_x))
+
+        is_bn_in_model = torch.nn.BatchNorm2d in [type(module) for name, module in quant_model.named_modules()]
+        self.unit_test.assertTrue(self.fold_applied is not is_bn_in_model)
+        self.unit_test.assertTrue(np.isclose(out_quant, out_float, atol=1e-6, rtol=1e-4).all())
+
+
+class BNForwardFoldingNet(torch.nn.Module):
```

Comment:
I would add a test with 2 bn, something like: **bn->conv->bn->relu** to check that it converted to **conv->relu**